src/engine/BoundsExplainer.cpp

```diff
@@ -0,0 +1,270 @@
+/*********************                                                        */
+/*! \file BoundsExplainer.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Omri Isac, Guy Katz
+ ** This file is part of the Marabou project.
+ ** Copyright (c) 2017-2022 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved. See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** [[ Add lengthier description here ]]
+ **/
+
+#include "BoundsExplainer.h"
+
+BoundsExplainer::BoundsExplainer( const unsigned varsNum, const unsigned rowsNum )
+	:_varsNum( varsNum )
+	,_rowsNum( rowsNum )
+	,_upperBoundsExplanations( _varsNum, std::vector<double>( 0 ) )
+	,_lowerBoundsExplanations( _varsNum,  std::vector<double>( 0 ) )
+{
+}
+
+BoundsExplainer::~BoundsExplainer()
+{
+	for ( unsigned i = 0; i < _varsNum; ++i )
+	{
+		_upperBoundsExplanations[i].clear();
+		_lowerBoundsExplanations[i].clear();
+	}
+
+	_upperBoundsExplanations.clear();
+	_lowerBoundsExplanations.clear();
+}
+
+unsigned BoundsExplainer::getRowsNum() const
+{
+	return _rowsNum;
+}
+
+unsigned BoundsExplainer::getVarsNum() const
+{
+	return _varsNum;
+}
+
+BoundsExplainer& BoundsExplainer::operator=( const BoundsExplainer& other )
```

Comment:
I'm pretty sure you can completely erase this function, too, and use the default. std::vectors, or our Vectors, know how to copy themselves.